internal/app/secretless/handlers/mysql/connection_details_test.go

```diff
@@ -0,0 +1,83 @@
+package mysql
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConnectionDetailsCorrectlyParsesConnectionDetails(t *testing.T) {
+	credentials := make(map[string][]byte)
+
+	credentials["host"] = []byte("myhost")
+	credentials["port"] = []byte("1234")
+	credentials["username"] = []byte("myusername")
+	credentials["password"] = []byte("mypassword")
+
+	expectedConnDetails := ConnectionDetails{
+		Host:     "myhost",
+		Port:     1234,
+		Username: "myusername",
+		Password: "mypassword",
+		Options:  make(map[string]string),
+	}
+
+	actualConnDetails, err := NewConnectionDetails(credentials)
+	assert.Nil(t, err)
+
+	assert.EqualValues(t, expectedConnDetails, *actualConnDetails)
+}
+
+func TestPortDefaultsTo3306IfNotProvided(t *testing.T) {
```

Comment:
I think `IfNotProvided` is redundant (implied by the definition of "default") and for the same magic number argument I noted in other comments we shouldn't have 3306 in the name.  

Suggested improvement: `TestDefaultPort`